jabgui/src/test/java/org/jabref/gui/externalfiles/AutoSetFileLinksUtilTest.java

```diff
@@ -132,12 +132,13 @@ void findAllAssociatedNotLinkedFilesInsteadOfTheFirstOne(@TempDir Path tempDir)
                 filePreferences,
                 autoLinkPrefs);
 
-        List<LinkedFile> matches = util.findAssociatedNotLinkedFiles(entry);
+        List<String> matchedFiles = util.findAssociatedNotLinkedFiles(entry)
+                                        .stream().map(LinkedFile::getLink).toList();
+        assertEquals(2, matchedFiles.size());
 
-        assertEquals(2, matches.size());
-        assertTrue(matches.stream().anyMatch(file ->
-                Path.of(newPath1String).equals(Path.of(file.getLink()))));
-        assertTrue(matches.stream().anyMatch(file ->
-                Path.of(newPath2String).equals(Path.of(file.getLink()))));
+        List<String> expected = List.of(newPath1String, newPath2String);
+        // findAssociatedNotLinkedFiles does not guarantee how the returned files are ordered
```

Comment:
Then we found a flaw, don't we?

1. Stabilize findAssociatedNotLinkedFiles to return the files in the order they appear in the BibEntry
2. Use `Collection` as return value

I prefer the former.

Reply:
I was about to comment

> The function that accumulates associated files says nothing about how the returned file is ordered, so I used the HashSet trick to check.

Since multiple files (with the same name) might be found here for one broken linked file in a BibEntry, it seems there is no way to arrange the found files according to their order in `BibEntry`.